MH-13381, Use Organization Identifier In Roles

lkiesow opened the pull request:
…instead of storing the whole organization and all its properties itself
since that causes a huge overhead when it comes to serialization of
users (and thus also workflows).

This patch uses a similar method to pull request #763 which means that this again does not need any migration of old data. They can still be used in their old form but new ones will not any longer include the whole organization.

Serialized roles now look like this (excerpt from a workflow instance XML):
```xml
<wf:workflow …>
  <wf:template>fast</wf:template>
  …
  <sec:creator>
    <sec:username>opencast_system_account</sec:username>
    <sec:name>System User</sec:name>
    <sec:provider>system</sec:provider>
    <sec:manageable>false</sec:manageable>
    <sec:roles>
      <sec:role>
        <sec:name>ROLE_ADMIN</sec:name>
        <sec:organizationid>mh_default_org</sec:organizationid>
        <sec:type>INTERNAL</sec:type>
      </sec:role>
      <sec:role>
        <sec:name>ROLE_SUDO</sec:name>
        <sec:organizationid>mh_default_org</sec:organizationid>
        <sec:type>INTERNAL</sec:type>
      </sec:role>
    </sec:roles>
    …
```

lkiesow committed:
MH-13381, Use Organization Identifier In Roles

…instead of storing the whole organization and all its properties itself
since that causes a huge overhead when it comes to serialization of
users (and thus also workflows).

staubesv commented on modules/common/src/main/java/org/opencastproject/security/api/JaxbRole.java:
```diff
@@ -45,10 +47,12 @@
   @XmlElement(name = "organization")
   protected JaxbOrganization organization;
 
+  @XmlElement(name = "organizationid")
```
Maybe organization_id or organizationId for sake of readability?

smarquard replied:
We have several precedents to choose from :-)

```
@XmlElement(name = "warning_state_trigger")
@XmlElement(name = "processingHost")
@XmlElement(name = "samplingrate")
```

but of these, camelCase seems to be the more common style used in our code base so far.

lkiesow replied:
In #763, I've switched to `organization-id` because of `execution-history` and `time-in-queue`, `fail-on-error`, `retry-strategy`, … it seemed to me that `-` was commonly used ;-D

I tend to use it here as well unless someone wants something else in particular.

staubesv replied:
For me, organization-id is fine.

smarquard replied:
Sounds good.

staubesv replied:
@lkiesow #763 is merged, so let's go for organization-id ...

smarquard commented:
I've tested this and PR #763 by merging both into our 6.x dev system and they have the desired effect and do not seem to have broken anything.

staubesv commented on modules/common/src/main/java/org/opencastproject/security/api/JaxbRole.java:
```diff
@@ -134,11 +157,15 @@ public String getDescription() {
    */
-  public Organization getOrganization() {
-    return organization;
+  public String getOrganizationId() {
```
Matter of taste thing: Personally, I would write

```
if (organizationId != null) {
  return organizationId;
} else if (organization != null) {
  return organization.getId();
} else {
  return null;
}
```

... but fine for me if you prefer the version without else if.

lkiesow replied:
I'll keep the current version.

staubesv reviewed (dismissed):
Please rename organizationid to organization-id as discussed. Besides that, LGTM.

lkiesow committed:
MH-13381, Use organization-id in XML

…in the same way we already use if for the reference in the workflow
instance XML.
